From 9c8a08c9c7a0f7ec591042abd5b4a0fb50c028f1 Mon Sep 17 00:00:00 2001
From: Nick Hudson <skrll@NetBSD.org>
Date: Mon, 28 Sep 2015 22:49:21 +0200
Subject: Increase BIOS_BSIZE.

libpciaccess expects a much bigger buffer when reading ROM.
Bump it to the (normal) IO space size.

Signed-off-by: Thomas Klausner <wiz@NetBSD.org>

diff --git a/src/s3v_driver.c b/src/s3v_driver.c
index d519dc8..e5dd2e4 100644
--- src/s3v_driver.c
+++ src/s3v_driver.c
@@ -316,7 +316,7 @@ s3virgeSetup(pointer module, pointer opts, int *errmaj, int *errmin)
 
 static unsigned char *find_bios_string(S3VPtr ps3v, int BIOSbase, char *match1, char *match2)
 {
-#define BIOS_BSIZE 1024
+#define BIOS_BSIZE 0x10000
 #define BIOS_BASE  0xc0000
 
    static unsigned char bios[BIOS_BSIZE];
-- 
cgit v0.10.2

